lookout/style/reporter.py

```diff
@@ -0,0 +1,146 @@
+            self._fs = os.path.join(self._tmpdir, "models")
+        os.makedirs(self._fs, exist_ok=True)
+        self._analyzer_context_manager = AnalyzerContextManager(
+            self.inspected_analyzer_type, db=self._database, fs=self._fs, init=False)
+        self._analyzer_context_manager.__enter__()
+        return self
+
+    def __exit__(self, exc_type=None, exc_val=None, exc_tb=None):
+        self._analyzer_context_manager.__exit__()
+        if self._tmpdir:
+            shutil.rmtree(self._tmpdir)
+
+    def run(self, dataset: Sequence[Dict[str, Any]]) -> Iterator[Dict[str, str]]:
+        """
+        Run report generation.
+
+        :param dataset: The dataset for the report generation. The format is a list of data rows. \
+                        The row is a Dictionary with a mapping from the column name to its content.
+
+        :return: Iterator through generated reports. Each Generated report is extended with the \
+                 corresponding row data from the dataset.
+        """
+        def _run(dataset) -> Iterator[Dict[str, str]]:
+            for index, row in enumerate(dataset):
+                self._log.info("processing %d / %d (%s)", index, len(dataset), row)
+                try:
+                    fixes = self._trigger_review_event(row)
+                    reports = self._generate_reports(row, fixes)
+                    reports.update(row)
+                    yield reports
+                except Exception:
+                    self._log.exception("Failed to generate report %d / %d (%s)",
+                                        index, len(dataset), row)
+
+        self._log.info("processing %d entries", len(dataset))
+        yield from self._finalize(_run(dataset))
+
+    @classmethod
+    def get_report_names(cls) -> Tuple[str, ...]:
+        """
+        Get all available report names.
+
+        :return: Tuple with report names.
+        """
+        raise NotImplementedError()
+
+    def _generate_reports(self, dataset_row: Dict[str, Any], fixes: Sequence[NamedTuple],
+                          ) -> Dict[str, str]:
+        """
+        Generate reports for the dataset row.
+
+        :param dataset_row: Dataset row which triggered the analyze method of the analyzer.
+        :param fixes: List of data provided by the analyze method of spied analyzer.
+        :return: Dictionary with report names as keys and report string as values.
+        """
+        raise NotImplementedError()
+
+    def _trigger_review_event(self, dataset_row: Dict[str, Any]) -> Sequence[NamedTuple]:
+        """
+        Trigger review event and convert provided comments to internal representation.
+
+        It is required to call `Reporter._analyzer_context_manager.review()` in this function with
+        arguments you need and convert provided comments to a Sequence of NamedTuple-s for the
+        report generation.
+
+        :param dataset_row: Dataset row which triggers the analyze method of the analyzer.
```

Comment:
```suggestion
        :param dataset_row: Dataset row which is passed to the `analyze()` method of the analyzer.
```

Reply:
it is not passed to `analyze()` because `analyze()` is triggered by `lookout_sdk.review()` and we cannot send there anything we want. I think about how to make this part clearer.